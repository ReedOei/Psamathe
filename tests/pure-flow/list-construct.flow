;

true --> var b : bool
[ ! bool ; b ] --> var x : list ! bool
true --> b
[ ! bool ; b ] --> x

