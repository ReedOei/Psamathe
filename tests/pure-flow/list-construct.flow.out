<generatedTop>
  <common>
    <typeEnv>
      .Map
    </typeEnv>
    <types>
      .Map
    </types>
  </common>
  <k>
    .
  </k>
  <functions>
    .Map
  </functions>
  <lookup>
    b |-> selected ( loc ( 0 ) , loc ( 0 ) )
    x |-> selected ( loc ( 2 ) , loc ( 2 ) )
  </lookup>
  <storage>
    loc ( 0 ) |-> resource ( bool , .List )
    loc ( 1 ) |-> resource ( bool , .List )
    loc ( 2 ) |-> resource ( table ( .Vars ) ! bool , ListItem ( loc ( 3 ) )
    ListItem ( loc ( 5 ) ) )
    loc ( 3 ) |-> resource ( bool , ListItem ( true ) )
    loc ( 4 ) |-> resource ( bool , .List )
    loc ( 5 ) |-> resource ( bool , ListItem ( true ) )
  </storage>
  <catchStack>
    .List
  </catchStack>
</generatedTop>
